packages/engine/lib/orchestrator/src/process/local.rs

```diff
@@ -29,17 +30,46 @@ pub struct LocalProcess {
 
 #[async_trait]
 impl process::Process for LocalProcess {
-    async fn exit_and_cleanup(mut self: Box<Self>) -> Result<()> {
-        self.child
-            .kill()
-            .await
-            .or_else(|e| match e.kind() {
-                // From `Child::kill` docs: Forces the child process to exit. If the child has
-                // already exited, an InvalidInput error is returned
-                std::io::ErrorKind::InvalidInput => Ok(()),
-                _ => Err(Report::new(e)),
-            })
-            .wrap_err("Could not kill the process")?;
+    async fn exit_and_cleanup(mut self: Box<Self>, experiment_id: ExperimentId) -> Result<()> {
+        // Leave some time for the engine to cleanup some resources
+        let hash_engine_result =
+            tokio::time::timeout(tokio::time::Duration::from_millis(500), self.child.wait()).await;
```

Comment:
We are doing this exact same thing at the main loop in `experiment.rs`. Why do we need this here again?

Reply:
We are? Where?

Reply:
We have a `sleep`-timeout of `self.config.wait_timeout` in line 289, and `engine_process.wait()` in line 298. It sets the `graceful_finish` to `false` if any of them is entered.